examples/port_filter.rs

```diff
@@ -0,0 +1,87 @@
+use bcc::Socket;
+use bcc::BPF;
+use clap::{App, Arg};
+use std::{thread, time};
+
+const DEFAULT_DURATION: u64 = 120; // Seconds
+
+pub fn configure_socket(socket_fd: i32) {
+    let mut flags = unsafe { libc::fcntl(socket_fd, libc::F_GETFL) };
+    flags = flags & !libc::O_NONBLOCK;
+    unsafe {
+        libc::fcntl(socket_fd, libc::F_SETFL, flags);
+    }
+}
+
+pub fn recv_loop(iface: String, fd: i32) {
+    loop {
+        let mut buf: [u8; 2048] = [0; 2048];
+        unsafe {
+            libc::read(fd, buf.as_mut_ptr() as *mut libc::c_void, 2048);
```

Comment:
Could we provide something that gives a better experience than this? It seems to go against the rest of this crate to return something that the user then needs to use libc and unsafe with.